src/map/lua/luautils.cpp

```diff
@@ -1629,7 +1597,7 @@ namespace luautils
     int32 OnEventFinish(CCharEntity* PChar, uint16 eventID, uint32 result)
     {
         //#TODO: move this to BCNM stuff when it's rewritten
-        if (PChar->PBCNM && PChar->PBCNM->won())
+        if (PChar->PBCNM && (PChar->PBCNM->won() || PChar->PBCNM->lost() || eventID == 32003)) // 32003 is the run-away event
```

Comment:
you can remove this, I fixed the bcnm loss message (and will be looking at run away shortly)